File: Johanns_Harp/Johanns_Harp.rb
# Johann's Harp solution in Ruby
# Rubberduck.tech + human-edited
# 2023-02-14

# Factorzation function
# start = starting number; factor; iterate = loop count
def factorize(start, factor, iterate)
  result = start
  iterate.times do
    result *= factor
  end
  return result
end

# Round the result to 6 decimal places
puts format("%.6f", factorize(1.0, 2.0 ** (1.0 / 12.0), 12))

File: Johanns_Harp/Johanns_Harp.nim
# Johann's Harp solution in Nim
# Rubberduck.tech + human-edited
# 2023-02-14

import std/strformat
import std/math

# Factorize 12 times, the 2^(1/12) increase each time
var product = 1.0
var factor = math.pow(2,1/12)
for i in 1..12:
    product *= factor

# Round the result to 6 decimal places
echo fmt"{product:.6f}"
